ims/einstein/iscsi/tgt.py

```diff
@@ -11,180 +12,154 @@
             raise iscsi_exceptions.StartFailedException()
 
     @log
     def stop_server(self):
-        '''
-        Need to check if this
-        :return:
-        '''
-        # arglist = self.arglist
-        # arglist.append("stop")
-        command = "service tgtd stop"
-        proc = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE,
-                                stderr=subprocess.PIPE)
-        output, err = proc.communicate(self.root_password + "\n")
-        logger.debug("Output = %s, Error = %s", output, err)
-        if self.show_status() is not 'Dead':
+        """
+        Stops the tgtd service
+
+        :return: None
+        """
+        try:
+            command = "service tgtd stop"
+            output = shell.call(command, sudo=True)
+            logger.debug("Output = %s", output)
+            if self.show_status() is not 'Dead':
+                raise iscsi_exceptions.StopFailedException()
+        except subprocess.CalledProcessError as e:
             raise iscsi_exceptions.StopFailedException()
 
     @log
     def restart_server(self):
-        '''
-        Again have to parse the output and send the values.
-        :return:
-        '''
+        """
+        Restarts the tgtd service
+
+        :return: None
+        """
         self.stop_server()
         self.start_server()
 
     @log
     def show_status(self):
-        # arglist = self.arglist
-        # arglist.append('status')
+        """
+        Returns the status of tgtd
+
+        :return: Running or Dead or Error as String
+        """
         command = "service tgtd status"
-        proc = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE,
-                                stderr=subprocess.PIPE)
-        status_string = proc.communicate(self.root_password + "\n")[0]
+        status_string = shell.call(command, sudo=True)
+        logger.debug("Output = %s", status_string)
         if 'active (running)' in status_string:
             return 'Running'
         elif 'inactive (dead)' in status_string:
             return 'Dead'
         # Have to check if there are any other states
         else:
-            return "Running in error state"
+            return "Error"
 
     def __generate_config_file(self, target_name):
-        config = open(self.TGT_ISCSI_CONFIG + target_name + ".conf", 'w')
+        """
+        Generates the config file in conf.d/
+
+        :param target_name: Target for which config file should be created
+        :return: None
+        """
+        config = open(
+            os.path.join(self.TGT_ISCSI_CONFIG, target_name + ".conf"), 'w')
         template_loc = os.path.abspath(
             os.path.join(os.path.dirname(os.path.abspath(__file__)), "..",
                          ".."))
-        for line in open(template_loc + "/tgt_target.temp", 'r'):
+        for line in open(os.path.join(template_loc, "tgt_target.temp"), 'r'):
             line = line.replace('${target_name}', target_name)
```

Comment:
No, I was asking about the green highlight for patch. Not sure, if its something to show difference though.